tests/lib/compiler-test-derive/src/ignores.rs

```diff
@@ -0,0 +1,195 @@
+            && self.compiler.as_ref().map_or(true, |val| val == compiler)
+            && (self.pattern_to_ignore == "*" || canonical_path.contains(&*self.pattern_to_ignore))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Ignores {
+    /// The canonical path, and the set of features
+    patterns: Vec<IgnorePattern>,
+}
+
+impl Ignores {
+    /// If the path matches any of the paths on the list
+    pub fn should_ignore(
+        &self,
+        os: &str,
+        arch: &str,
+        engine: &str,
+        compiler: &str,
+        canonical_path: &str,
+    ) -> bool {
+        self.patterns.iter().any(|p| {
+            // println!(" -> {:?}", p);
+            p.should_ignore(os, arch, engine, compiler, canonical_path)
+        })
+    }
+
+    pub fn should_ignore_host(&self, engine: &str, compiler: &str, canonical_path: &str) -> bool {
+        self.should_ignore(
+            CFG_TARGET_OS,
+            CFG_TARGET_ARCH,
+            engine,
+            compiler,
+            canonical_path,
+        )
+    }
+
+    /// Build a Ignore structure from a file path
+    pub fn build_from_path(path: PathBuf) -> Ignores {
+        let file = File::open(path).unwrap();
+        let reader = BufReader::new(file);
+        let mut patterns = Vec::new();
+
+        for line in reader.lines() {
+            let line = line.unwrap();
+            // If the line has a `#` we discard all the content that comes after
+            let line = if line.contains('#') {
+                let l: Vec<&str> = line.split('#').collect();
+                l.get(0).unwrap().to_string()
+            } else {
+                line
+            };
+
+            let line = line.trim().to_string();
+
+            // If the lines contains ` ` it means the test should be ignored
+            // on the features exposed
+            if line.contains(" ") {
+                let l: Vec<&str> = line.splitn(2, " ").collect();
+                let mut os: Option<String> = None;
+                let mut arch: Option<String> = None;
+                let mut engine: Option<String> = None;
+                let mut compiler: Option<String> = None;
+                for alias in l.get(0).unwrap().trim().split("+") {
+                    match alias {
+                        "aarch64" | "x86" | "x64" => {
+                            arch = Some(alias.to_string());
+                        }
+                        "windows" | "macos" | "linux" => {
+                            os = Some(alias.to_string());
+                        }
+                        "jit" | "native" => {
+                            engine = Some(alias.to_string());
+                        }
+                        "cranelift" | "llvm" | "singlepass" => {
+                            compiler = Some(alias.to_string());
+                        }
+                        other => {
+                            panic!("Alias {:?} not currently supported", other);
```

Comment:
Minor point, it might be nice to keep a counter of which line number we're looking at so we can report it in errors. Not high priority, but it's a nice thing to have

Reply:
Done!